src/SFA.DAS.PR.Data/Repositories/EmployerRelationshipsReadRepository.cs

```diff
@@ -19,6 +19,9 @@ public class EmployerRelationshipsReadRepository(IProviderRelationshipsDataConte
             .Include(acc => acc.AccountLegalEntities)
                 .ThenInclude(ale => ale.Requests)
                     .ThenInclude(r => r.PermissionRequests)
+            .Include(acc => acc.AccountLegalEntities)
```

Comment:
I am interested in seeing what does this query look like, it may pose performance issue if it does not have the right indexes.

Reply:
Generated query looks like follows: 
```
exec sp_executesql N'SELECT [t].[Id], [t].[Created], [t].[HashedId], [t].[Name], [t].[PublicHashedId], [t].[Updated], [t1].[Id], [t1].[AccountId], [t1].[Created], [t1].[Deleted], [t1].[Name], [t1].[PublicHashedId], [t1].[Updated], [t1].[Id0], [t1].[AccountLegalEntityId], [t1].[AccountProviderId], [t1].[Created0], [t1].[Updated0], [t1].[Id00], [t1].[Ukprn], [t1].[Id1], [t1].[AccountProviderLegalEntityId], [t1].[Operation], [t1].[AccountId0], [t1].[Created00], [t1].[ProviderUkprn], [t1].[Created1], [t1].[Name0], [t1].[Updated00], [t1].[Id2], [t1].[AccountLegalEntityId0], [t1].[ActionedBy], [t1].[EmployerAORN], [t1].[EmployerContactEmail], [t1].[EmployerContactFirstName], [t1].[EmployerContactLastName], [t1].[EmployerOrganisationName], [t1].[EmployerPAYE], [t1].[RequestType], [t1].[RequestedBy], [t1].[RequestedDate], [t1].[Status], [t1].[Ukprn0], [t1].[UpdatedDate], [t1].[Ukprn00], [t1].[Id01], [t1].[Operation0], [t1].[RequestId], [t1].[Created2], [t1].[Name1], [t1].[Updated1]
FROM (
    SELECT TOP(1) [a].[Id], [a].[Created], [a].[HashedId], [a].[Name], [a].[PublicHashedId], [a].[Updated]
    FROM [Accounts] AS [a]
    WHERE [a].[Id] = @__accountId_0
) AS [t]
LEFT JOIN (
    SELECT [a0].[Id], [a0].[AccountId], [a0].[Created], [a0].[Deleted], [a0].[Name], [a0].[PublicHashedId], [a0].[Updated], [t0].[Id] AS [Id0], [t0].[AccountLegalEntityId], [t0].[AccountProviderId], [t0].[Created] AS [Created0], [t0].[Updated] AS [Updated0], [t0].[Id0] AS [Id00], [t0].[Ukprn], [t0].[Id1], [t0].[AccountProviderLegalEntityId], [t0].[Operation], [t0].[AccountId] AS [AccountId0], [t0].[Created0] AS [Created00], [t0].[ProviderUkprn], [t0].[Created1], [t0].[Name] AS [Name0], [t0].[Updated0] AS [Updated00], [t2].[Id] AS [Id2], [t2].[AccountLegalEntityId] AS [AccountLegalEntityId0], [t2].[ActionedBy], [t2].[EmployerAORN], [t2].[EmployerContactEmail], [t2].[EmployerContactFirstName], [t2].[EmployerContactLastName], [t2].[EmployerOrganisationName], [t2].[EmployerPAYE], [t2].[RequestType], [t2].[RequestedBy], [t2].[RequestedDate], [t2].[Status], [t2].[Ukprn] AS [Ukprn0], [t2].[UpdatedDate], [t2].[Ukprn0] AS [Ukprn00], [t2].[Id0] AS [Id01], [t2].[Operation] AS [Operation0], [t2].[RequestId], [t2].[Created] AS [Created2], [t2].[Name] AS [Name1], [t2].[Updated] AS [Updated1]
    FROM [AccountLegalEntities] AS [a0]
    LEFT JOIN (
        SELECT [a1].[Id], [a1].[AccountLegalEntityId], [a1].[AccountProviderId], [a1].[Created], [a1].[Updated], [a2].[Id] AS [Id0], [p].[Ukprn], [p0].[Id] AS [Id1], [p0].[AccountProviderLegalEntityId], [p0].[Operation], [a2].[AccountId], [a2].[Created] AS [Created0], [a2].[ProviderUkprn], [p].[Created] AS [Created1], [p].[Name], [p].[Updated] AS [Updated0]
        FROM [AccountProviderLegalEntities] AS [a1]
        INNER JOIN [AccountProviders] AS [a2] ON [a1].[AccountProviderId] = [a2].[Id]
        INNER JOIN [Providers] AS [p] ON [a2].[ProviderUkprn] = [p].[Ukprn]
        LEFT JOIN [Permissions] AS [p0] ON [a1].[Id] = [p0].[AccountProviderLegalEntityId]
    ) AS [t0] ON [a0].[Id] = [t0].[AccountLegalEntityId]
    LEFT JOIN (
        SELECT [r].[Id], [r].[AccountLegalEntityId], [r].[ActionedBy], [r].[EmployerAORN], [r].[EmployerContactEmail], [r].[EmployerContactFirstName], [r].[EmployerContactLastName], [r].[EmployerOrganisationName], [r].[EmployerPAYE], [r].[RequestType], [r].[RequestedBy], [r].[RequestedDate], [r].[Status], [r].[Ukprn], [r].[UpdatedDate], [p1].[Ukprn] AS [Ukprn0], [p2].[Id] AS [Id0], [p2].[Operation], [p2].[RequestId], [p1].[Created], [p1].[Name], [p1].[Updated]
        FROM [Requests] AS [r]
        INNER JOIN [Providers] AS [p1] ON [r].[Ukprn] = [p1].[Ukprn]
        LEFT JOIN [PermissionRequests] AS [p2] ON [r].[Id] = [p2].[RequestId]
    ) AS [t2] ON [a0].[Id] = [t2].[AccountLegalEntityId]
    WHERE [a0].[Deleted] IS NULL
) AS [t1] ON [t].[Id] = [t1].[AccountId]
ORDER BY [t].[Id], [t1].[Id], [t1].[Id0], [t1].[Id00], [t1].[Ukprn], [t1].[Id1], [t1].[Id2], [t1].[Ukprn00]',N'@__accountId_0 bigint',@__accountId_0=76433
```

We need to refactor this else we will see performance issues with employers who work with many providers and have multiple legal entities.